### PR TITLE
feat: custom assistant avatars

### PR DESCRIPTION
**Pull request by fontanierh**
⚠️ before it's usable, I will set the en var for the uploads bucket in production

Allows users to upload & crop their custom avatars for their assistants

Screenshot 2023-09-26 at 14 19 53
Screenshot 2023-09-26 at 14 20 16
Screenshot 2023-09-26 at 14 20 24

### CODE REVIEW

**Review by spolu — CHANGES_REQUESTED** [review]
Looks great! Minor UX comment

File: front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                visual={<img src={url} />}
+                clickable={available}
+              />
+            </div>
+          ))}
+        </div>
+      )}
+      {currentTab === "upload" && src && (
+        <div className="pt-8">
+          <ReactCrop crop={crop} aspect={1} onChange={(_, pC) => setCrop(pC)}>
```
Comment (spolu) [inline front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx:207]: Should be centered? :grimacing:

Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx:207]: 😬

Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx]: Fixed

File: front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+              label: "Upload your own",
+              current: currentTab === "upload",
+              icon: ArrowUpOnSquareIcon,
+            },
+          ]}
+          onTabClick={(tab) => {
+            const isUploadTab = tab.startsWith("Upload");
+            if (isUploadTab) {
+              // triggers file selection. once file is selected, tab will change to `upload`
+              fileInputRef?.current?.click();
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx]: fixed

[the remark this answers is not in the document]
Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx]: Screenshot 2023-09-26 at 15 25 48

like this

**Review by spolu — DISMISSED** [review]

File: front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx (hunk 1 above)
Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilderAvatarPicker.tsx]: Screenshot 2023-09-26 at 15 26 46

**Comment by Duncid** [comment]
Ok looking good.
Why not make it a full page modal?.

**Comment by fontanierh** [comment]
> Why not make it a full page modal?

We def could, I would decorelate the 2 however, we have this task related to improving the avatar picker UI https://github.com/orgs/dust-tt/projects/3?pane=issue&itemId=39382654 (full screen modal + auto-resize avatars)

**Pull request merged**